packages/jellyfish-api-core/__tests__/category/masternode/setGov.locks.token.test.ts

```diff
@@ -0,0 +1,585 @@
+    }
+
+    // Lock and unlock loan tokens in the same block
+    await testing.rpc.masternode.setGov({ ATTRIBUTES: { [`v0/locks/token/${tslaId}`]: 'true' } })
+    await testing.rpc.masternode.setGov({ ATTRIBUTES: { [`v0/locks/token/${tslaId}`]: 'false' } })
+    await testing.generate(1)
+
+    {
+      const attributes = await testing.rpc.masternode.getGov('ATTRIBUTES')
+      expect(
+        attributes.ATTRIBUTES[`v0/locks/token/${tslaId}`] === 'true' ||
+        attributes.ATTRIBUTES[`v0/locks/token/${tslaId}`] === 'false'
+      ).toStrictEqual(true)
+    }
+
+    // Unlock and lock loan tokens in the same block
+    await testing.rpc.masternode.setGov({ ATTRIBUTES: { [`v0/locks/token/${tslaId}`]: 'false' } })
+    await testing.rpc.masternode.setGov({ ATTRIBUTES: { [`v0/locks/token/${tslaId}`]: 'true' } })
+    await testing.generate(1)
+
+    {
+      const attributes = await testing.rpc.masternode.getGov('ATTRIBUTES')
+      expect(
+        attributes.ATTRIBUTES[`v0/locks/token/${tslaId}`] === 'true' ||
+        attributes.ATTRIBUTES[`v0/locks/token/${tslaId}`] === 'false'
+      ).toStrictEqual(true)
+    }
+
+    // 2 lock loan tokens in the same block
+    await testing.rpc.masternode.setGov({ ATTRIBUTES: { [`v0/locks/token/${tslaId}`]: 'true' } })
+    await testing.rpc.masternode.setGov({ ATTRIBUTES: { [`v0/locks/token/${tslaId}`]: 'true' } })
+    await testing.generate(1)
+
+    {
+      const attributes = await testing.rpc.masternode.getGov('ATTRIBUTES')
+      expect(attributes.ATTRIBUTES[`v0/locks/token/${tslaId}`]).toStrictEqual('true')
+    }
+
+    // 2 unlock loan tokens in the same block
+    await testing.rpc.masternode.setGov({ ATTRIBUTES: { [`v0/locks/token/${tslaId}`]: 'false' } })
+    await testing.rpc.masternode.setGov({ ATTRIBUTES: { [`v0/locks/token/${tslaId}`]: 'false' } })
+    await testing.generate(1)
+
+    {
+      const attributes = await testing.rpc.masternode.getGov('ATTRIBUTES')
+      expect(attributes.ATTRIBUTES[`v0/locks/token/${tslaId}`]).toStrictEqual('false')
+    }
+
+    // Lock invalid loan token string
+    await testing.rpc.masternode.setGov({ ATTRIBUTES: { 'v0/locks/token/abc': 'true' } })
+    await testing.generate(1)
+
+    {
+      const attributes = await testing.rpc.masternode.getGov('ATTRIBUTES')
+      expect(attributes.ATTRIBUTES['v0/locks/token/abc']).toBeUndefined()
+    }
+
+    // Unlock invalid loan token string
+    await testing.rpc.masternode.setGov({ ATTRIBUTES: { 'v0/locks/token/abc': 'false' } })
+    await testing.generate(1)
+
+    {
+      const attributes = await testing.rpc.masternode.getGov('ATTRIBUTES')
+      expect(attributes.ATTRIBUTES['v0/locks/token/abc']).toBeUndefined()
+    }
+  })
+
+  it('should update loan token if loan token is unlocked', async () => {
+    // Unlock loan token
+    await testing.rpc.masternode.setGov({ ATTRIBUTES: { [`v0/locks/token/${tslaId}`]: 'false' } })
```

Comment:
lock the token first. That will allow the system to do its thing. then unlock, and check the functionality is back to normal. 
with the current way, you only check one part of the flow.

Reply:
Lock token will provide a negative scenario. See the end of this file

Reply:
We need to put those for exception in one test item